### PR TITLE
Balder/summary.numthreads 2 backgroundnumthreads rebased 1

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge or @geirst PR

### CODE REVIEW

**Review by toregge — APPROVED** [review]
LGTM

**Comment by baldersheim** [comment]
I will merge this one now. If we find a better name, I will update after the weekend. Documentation will also be updated then.

File: config-model/src/main/java/com/yahoo/vespa/model/search/NodeFlavorTuning.java [hunk 1]
```diff
@@ -27,7 +27,7 @@ public void getConfig(ProtonConfig.Builder builder) {
         setHwInfo(builder);
         tuneDiskWriteSpeed(builder);
         tuneDocumentStoreMaxFileSize(builder.summary.log);
-        tuneDocumentStoreNumThreads(builder.summary.log);
+        tuneDocumentStoreNumThreads(builder.background);
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/vespa/model/search/NodeFlavorTuning.java]: Consider rename function to reflect which builder is used

File: config-model/src/test/java/com/yahoo/vespa/model/search/NodeFlavorTuningTest.java [hunk 2]
```diff
@@ -97,7 +97,7 @@ private static void assertFlushStrategyMemory(long expMemoryBytes, int memoryGb)
     }
 
     private static void assertDocumentStoreNumThreads(int numThreads, double numCores) {
```
Comment (geirst) [inline config-model/src/test/java/com/yahoo/vespa/model/search/NodeFlavorTuningTest.java]: Consider rename function to reflect new setting being tested

File: searchcore/src/vespa/searchcore/config/proton.def [hunk 3]
```diff
@@ -376,6 +378,11 @@ visit.ignoremaxbytes bool default=true
 ## When set to 0 (default) we use 1 separate thread per document database.
 initialize.threads int default = 0
 
+## Number of worker threads doing background compaction/compression tasks.
+## They all live i a shared thread pool.
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/config/proton.def]: Typo

File: searchcore/src/vespa/searchcore/proton/server/proton.cpp [hunk 4]
```diff
@@ -85,12 +85,24 @@ diskMemUsageSamplerConfig(const ProtonConfig &proton, const HwInfo &hwInfo)
             hwInfo);
 }
 
+size_t
+deriveBackgroundThreads(const ProtonConfig & proton) {
+    size_t threads = std::thread::hardware_concurrency();
+    if (proton.background.threads != 0) {
+        threads = proton.background.threads;
+    }
+    threads = std::max(threads, size_t(proton.summary.log.numthreads));
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/proton.cpp]: Shouldn't we use 'proton.background.threads' here?

**Comment by geirst** [comment]
@baldersheim please have a look at my comments for next PR

File: searchcore/src/vespa/searchcore/proton/server/proton.cpp (hunk 4 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/proton.cpp]: No, that has been taken into account above.

**Pull request merged**